docs/example_python.md

```diff
@@ -1,30 +1,194 @@
+*The server is up! (Click the screenshot to see an interactive view.)*
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Tilt can run commands locally, so that in addition to running things in your cluster, you can execute existing scripts or arbitrary shell commands on your own machine.
+
+In this example, we use [`local_resource`](local_resource.html), which lets you locally run scripts, shell code, or servers. We add a `local_resource` to our
+[Tiltfile](https://github.com/windmilleng/tilt-example-python/blob/master/1-measured/Tiltfile)
+that records the current time, then kicks off an update. We've also modified our server itself to
+read that start time and print the time elapsed.
+
+```python
+# Records the current time, then kicks off a server update.
+# Normally, you would let Tilt do deploys automatically, but this
+# shows you how to set up a custom workflow that measures it.
+local_resource(
+  'deploy',
+  'date +%s > start-time.txt'
+)
+…
+k8s_resource('example-python', port_forwards=5000, resource_deps=['deploy'])
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the command that it runs.
+
+See that button next to the `deploy` resource?
+![Button to trigger a deploy](assets/docimg/example-python-2-deploy-button.png)
+
+Let's click it and see what happens!
+
+[![Result of clicking the button  on the “deploy” resource](assets/docimg/example-python-3-naive-deploy.png)](https://cloud.tilt.dev/snapshot/AcyUjOYLaeKVxQ1XZjk=){:.is-image}
+*Clicking the button triggers the "deploy" local_resource, which in turn kicks off an update to the server. (Click the screenshot to see an interactive view.)*
+
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 10-11s |
+
+Can we do better?
+
+## Step 2: Let's Optimize It
+
+When we make a change to a file, we currently have to build an image, deploy new Kubernetes configs,
+and wait for Kubernetes to schedule the pod.
+
+With Tilt, we can skip all of these steps, and instead [live-update](https://docs.tilt.dev/live_update_tutorial.html) the pod in place.
+
+[Our new Tiltfile](https://github.com/windmilleng/tilt-example-html/blob/master/2-recommended/Tiltfile) contains the following new code:
+
+```python
+# Add live_update rules to our docker_build
+congrats = "🎉 Congrats, you ran a live_update! 🎉"
+docker_build('example-python-image', '.', live_update=[
+    sync('.', '/app'),
+    run('cd /app && pip install -r requirements.txt', trigger='/app/requirements.txt'),
+
+    # if all that changed was start-time.txt, make sure the server
+    # reloads so that it will reflect the new startup time
+    run('touch /app/app.py', trigger='start-time.txt'),
+
+    # add a congrats message!
+    run('sed -i "s/Hello cats!/{}/g" /app/templates/index.html'.format(congrats)),
+])
+```
+
+We've added a new parameter to `docker_build()` with four `live_update` steps that are, in order:
```

Comment:
I also saw your example code add debug=True to the Flask app. is that worth calling out here, to make it clear that Flask is doing the live-reloading?

Though TBH, my understanding of Flask is that you're supposed to use `flask run` when you're in dev mode. I think we would recommend that people use entrypoint='flask run' in their Tiltfile, rather than hard-coding debug mode in the app, right?

Reply:
ooh good call, let me look into that.